cpp/src/strings/regex/regcomp.cpp

```diff
@@ -292,114 +333,90 @@ class regex_parser {
   {
     int quoted = nextc(yy);
     if (quoted) {
-      // treating all quoted numbers as Octal, since we are not supporting backreferences
-      if (yy >= '0' && yy <= '7') {
-        yy          = yy - '0';
-        auto c      = *exprp;
-        auto digits = 1;
-        while (c >= '0' && c <= '7' && digits < 3) {
-          yy = (yy << 3) | (c - '0');
-          c  = *(++exprp);
-          ++digits;
+      switch (yy) {
+        case 't': yy = '\t'; break;
+        case 'n': yy = '\n'; break;
+        case 'r': yy = '\r'; break;
+        case 'a': yy = 0x07; break;
+        case 'f': yy = 0x0C; break;
+        case '0' ... '7': {
+          yy = handle_octal(yy);
+          break;
         }
-        return CHAR;
-      } else {
-        switch (yy) {
-          case 't': yy = '\t'; break;
-          case 'n': yy = '\n'; break;
-          case 'r': yy = '\r'; break;
-          case 'a': yy = 0x07; break;
-          case 'f': yy = 0x0C; break;
-          case '0': yy = 0; break;
-          case 'x': {
-            char32_t a = *exprp++;
-            char32_t b = *exprp++;
-            yy         = 0;
-            if (a >= '0' && a <= '9')
-              yy += (a - '0') << 4;
-            else if (a >= 'a' && a <= 'f')
-              yy += (a - 'a' + 10) << 4;
-            else if (a >= 'A' && a <= 'F')
-              yy += (a - 'A' + 10) << 4;
-            if (b >= '0' && b <= '9')
-              yy += b - '0';
-            else if (b >= 'a' && b <= 'f')
-              yy += b - 'a' + 10;
-            else if (b >= 'A' && b <= 'F')
-              yy += b - 'A' + 10;
+        case 'x': {
+          yy = handle_hex();
+          break;
+        }
+        case 'w': {
```

Comment:
Everything after this point in this file is just a reformat (shift-left) by clang-format as a result of the if-else removal above.